src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -249,6 +250,7 @@ trait ScalaSettings extends AbsScalaSettings
   val Ydelambdafy        = ChoiceSetting  ("-Ydelambdafy", "strategy", "Strategy used for translating lambdas into JVM code.", List("inline", "method"), "method")
   val YmacroClasspath = PathSetting       ("-Ymacro-classpath", "The classpath used to reflectively load macro implementations, default is the compilation classpath.", "")
 
+  val YjarFactory = StringSetting   ("-YjarFactory", "classname", "factory for jar files", classOf[DefaultJarFactory].getName)
```

Comment:
Perhaps rather than `-Y` for this, we could just have a var `backend.jarFactory` field that compiler plugins could mutate on initialization.

Reply:
Settings can be `.internal` if using a setting helps with decoupling.

The bigger technical hurdle is keeping the equals signs vertically aligned.

Reply:
Having  magic field that plugins could manipulate but cant cooperate on seem a poor path. Who wins it they both want to specify one?